modules/RTC/RTCUtils.js

```diff
@@ -321,6 +321,7 @@ function pollForAvailableMediaDevices() {
  * @emits RTCEvents.DEVICE_LIST_CHANGED
  */
 function onMediaDevicesListChanged(devices) {
+    var initialDeviceList = currentlyAvailableMediaDevices.length == 0;
```

Comment:
I don't think this is the best way to do this. The more correct way is in https://github.com/jitsi/lib-jitsi-meet/pull/139.
Let me explain the problem. The code the polls for ```enumerateDevices()``` is a workaround until ```"devicechange"``` event is supported in browsers. And if in future we will switch to that event natively, it won't compare old and new device, it will just fire and that's it. And with your new implementation we will just loose the first event.